driver/ni/ai_channels.h

```diff
@@ -0,0 +1,1459 @@
+        if (type == "K") return DAQmx_Val_K_Type_TC;
+        if (type == "N") return DAQmx_Val_N_Type_TC;
+        if (type == "R") return DAQmx_Val_R_Type_TC;
+        if (type == "S") return DAQmx_Val_S_Type_TC;
+        if (type == "T") return DAQmx_Val_T_Type_TC;
+        if (type == "B") return DAQmx_Val_B_Type_TC;
+        if (type == "E") return DAQmx_Val_E_Type_TC;
+        parser.field_err("channels." + name,
+                         "Invalid thermocouple type:" + type +
+                         ". Defaulting to J Type.");
+        return DAQmx_Val_J_Type_TC;
+    }
+
+    int32_t getCJCSource(const std::string &source, config::Parser &parser) {
+        if (source == "BuiltIn") return DAQmx_Val_BuiltIn;
+        if (source == "ConstVal") return DAQmx_Val_ConstVal;
+        if (source == "Chan") return DAQmx_Val_Chan;
+        parser.field_err("channels." + this->name,
+                         "Invalid thermocouple cjc source:" + source +
+                         ". Defaulting to BuiltIn.");
+        return DAQmx_Val_BuiltIn;
+    }
+
+    explicit Thermocouple(config::Parser &parser, TaskHandle task_handle,
+                          const std::string &name)
+        : Analog(parser, task_handle, name),
+          thermocoupleType(getType(parser.required<std::string>("thermocouple_type"),
+                                   parser)),
+          cjcSource(getCJCSource(parser.required<std::string>("cjc_source"), parser)),
+          cjcVal(parser.required<double>("cjc_val")) {
+        LOG(INFO) << "Thermocouple created with name: " << name;
+    }
+
+    //cjcChannel(parser.required<std::string>("cjc_channel")) {} FIXME: this property should be take form console
+    int32 createNIChannel() override {
+        return ni::NiDAQmxInterface::CreateAIThrmcplChan(
+            this->task_handle,
+            this->name.c_str(),
+            "",
+            this->min_val,
+            this->max_val,
+            this->units,
+            this->thermocoupleType,
+            this->cjcSource,
+            this->cjcVal,
+            ""
+        );
+    }
+};
+
+class TemperatureBuiltInSensor final : public Analog {
+public:
+    explicit TemperatureBuiltInSensor(config::Parser &parser, TaskHandle task_handle,
+                                      const std::string &name) {
+        this->units = ni::UNITS_MAP.at(parser.required<std::string>("units"));
+        this->task_handle = task_handle;
+        size_t pos = name.find("/");
+        this->name = name.substr(0, pos) + "/_boardTempSensor_vs_aignd";
+    }
+
+    int32 createNIChannel() override {
+        return ni::NiDAQmxInterface::CreateAITempBuiltInSensorChan(
+            this->task_handle,
+            this->name.c_str(),
+            "",
+            this->units
+        );
+    }
+};
+
+class ThermistorIEX final : public Analog {
+public:
+    int32_t resistance_config;
+    CurrentExcitationConfig excitation_config;
+    double a;
```

Comment:
Out of curiosity, do these fields need to be public? Not a big deal, but if all we're doing is calling `createNIChannel`, then I think these fields could be private

Reply:
some of the fields have to be public (e.g. current channel types need to be because currentRMS inherits from it). but i updated it so any possible ones are private